File: apps/backend/src/events/events.go
package events

const (
	// Client emitted events

	Connect          = "connect"
	CreateRoom       = "create_room"
	JoinRoom         = "join_room"
	LeaveRoom        = "leave_room"
	StartGame        = "start_game"
	PlayerMove       = "player_move"
	PlayerAttack     = "player_attack"
	GameState        = "game_state"
	GameEnd          = "game_end"
	PlayerDisconnect = "player_disconnect"
	GetRoomState     = "get_room_state"
	MovePlayer       = "move_player"

	// Server emitted events, mainly broadcasts

	PlayerJoined = "player_joined"
	PlayerLeft   = "player_left"
	GameLoop     = "game_loop"
	GameStarted  = "game_started"
)

File: apps/frontend/src/lib/constants/events.ts
export const EVENTS = {
	CONNECT: 'connect',
	CREATE_ROOM: 'create_room',
	JOIN_ROOM: 'join_room',
	LEAVE_ROOM: 'leave_room',
	GET_ROOM_STATE: 'get_room_state',
	PLAYER_JOINED: 'player_joined',
	PLAYER_LEFT: 'player_left',
	START_GAME: 'start_game',
	GAME_STARTED: 'game_started',
	GAME_LOOP: 'game_loop',
	MOVE_PLAYER: 'move_player'
} as const;

export type EventKey = keyof typeof EVENTS;
export type EventName = (typeof EVENTS)[EventKey];
